test/widgets/action_sheet_test.dart

```diff
@@ -93,6 +97,286 @@ void main() {
+        child: const InboxPage()));
+      await tester.pump();
+
+      await tester.longPress(find.text(topic));
+      // sheet appears onscreen; default duration of bottom-sheet enter animation
+      await tester.pump(const Duration(milliseconds: 250));
+      check(find.byType(BottomSheet)).findsOne();
+    });
+
+    testWidgets('show from app bar', (tester) async {
+      await prepare();
+      connection.prepare(json: eg.newestGetMessagesResult(
+        foundOldest: true, messages: [message]).toJson());
+      await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+        child: MessageListPage(
+          initNarrow: TopicNarrow(channel.streamId, topic))));
+      // global store, per-account store, and message list get loaded
+      await tester.pumpAndSettle();
+
+      await tester.longPress(find.byType(ZulipAppBar));
+      // sheet appears onscreen; default duration of bottom-sheet enter animation
+      await tester.pump(const Duration(milliseconds: 250));
+      check(find.byType(BottomSheet)).findsOne();
+    });
+
+    testWidgets('show from recipient header', (tester) async {
+      await prepare();
+      connection.prepare(json: eg.newestGetMessagesResult(
+        foundOldest: true, messages: [message]).toJson());
+      await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+        child: const MessageListPage(initNarrow: CombinedFeedNarrow())));
+      // global store, per-account store, and message list get loaded
+      await tester.pumpAndSettle();
+
+      await tester.longPress(find.descendant(
+        of: find.byType(RecipientHeader), matching: find.text(topic)));
+      // sheet appears onscreen; default duration of bottom-sheet enter animation
+      await tester.pump(const Duration(milliseconds: 250));
+      check(find.byType(BottomSheet)).findsOne();
+    });
+  });
+
+  group('UserTopicUpdateButton', () {
+    late ZulipStream channel;
+    late String topic;
+
+    Finder findButton({UserTopicVisibilityPolicy? from, required UserTopicVisibilityPolicy to}) {
+      return find.byWidgetPredicate((widget) {
+        return widget is UserTopicUpdateButton
+          && widget.currentVisibilityPolicy == from
+          && widget.newVisibilityPolicy == to;
+      });
+    }
+
+    final mute =                 findButton(to:   UserTopicVisibilityPolicy.muted);
+    final unmute =               findButton(from: UserTopicVisibilityPolicy.muted,
+                                            to:   UserTopicVisibilityPolicy.none);
+    final unmuteInMutedChannel = findButton(to:   UserTopicVisibilityPolicy.unmuted);
+    final follow =               findButton(to:   UserTopicVisibilityPolicy.followed);
+    final unfollow =             findButton(from: UserTopicVisibilityPolicy.followed,
+                                            to:   UserTopicVisibilityPolicy.none);
+
+    Future<void> setupToTopicActionSheet(WidgetTester tester, {
+      required bool? isChannelMuted,
```

Comment:
> Let's be more explicit about what null means for isChannelMuted. I think it's supposed to mean that the self-user isn't subscribed to the channel (right?), but that isn't clear from test-failure output—

Yeah. Updated the test description and leave the topic names as-is, because the description is what shows up when a test fails.